appveyor.yml

```diff
@@ -48,43 +48,36 @@ install:
 - cmd: |-
     cd scripts
-    C:\\Python35\\python set_version_to.py -v %APPVEYOR_BUILD_VERSION% -t %APPVEYOR_BUILD_VERSION%
+    C:\\Python311\\python set_version_to.py -v %APPVEYOR_BUILD_VERSION% -t %APPVEYOR_BUILD_VERSION%
     cd ..
     git --version
 
 
 # to run your custom scripts instead of automatic MSBuild
 build_script:
 - ps: |
-    $isBuildingPR = $false;
-    # if building a temporary merge with master, soft reset to the PR commit so the build contains the PR's hash instead of the merge-commit's hash
+    $buildArgs = '/p:ContinuousIntegrationBuild=true';
+    # if building a temporary merge with master, override the hash with the PR commit's so the build contains the PR's hash instead of the merge-commit's hash
     if ($env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT -and ($env:APPVEYOR_REPO_COMMIT -ne $env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT))
     {
-        git reset --soft "$env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT" --
-        $isBuildingPR = $true;
+        $gitCommit = $(git rev-parse --short $env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT)
+        $buildArgs +=  " /p:GitCommit=$gitCommit /p:GitSha=$($env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT)"
     }
+- ps: |
+    # add to index, to adjust crlf configured in .gitattributes
+    git -c core.autocrlf=false add -A
+    if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
 - ps: |
     # build VC++
     dotnet build .\scripts\native.proj -c Release --verbosity q --nologo /bl:.\artifacts\log\native.binlog
     if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
 - ps: |
     # build .NET
-    dotnet build -c Release --verbosity q --nologo /bl:.\artifacts\log\build.binlog /p:ContinuousIntegrationBuild=true
+    $cmd = "dotnet build -c Release --verbosity q --nologo /bl:.\artifacts\log\build.binlog $buildArgs"
+    iex $cmd
     if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
-- ps: |
-    # if we have reset above, we need to restore the merge commit, otherwise the loc verification step will fail
-    # there is no need to have the same commit hash, but there must be no pending changes before the loc verification
-    # refer to https://github.com/gitextensions/gitextensions/issues/7979
-    if ($isBuildingPR) {
-        git config user.email "gitextensions@github.com"
-        git config user.name "Git Extensions"
-        git -c core.autocrlf=false commit -a -m "restore merge commit" | Out-Null
-        # ignore warning about line-endings conversion
-        $global:LASTEXITCODE = 0
-    }
 - ps: |
     # Verify that new strings (if any) have been processed and ready for localisation
-    # it would be nice to run '.\cibuild.cmd -loc -logFileName localise.binlog /p:NoBuild=true' but it doesn't work without `-build` switch :\
     Push-Location .\GitExtensions
     dotnet msbuild /p:Configuration=Release /t:_UpdateEnglishTranslations /p:RunTranslationApp=true /p:ContinuousIntegrationBuild=true /v:m /bl:..\artifacts\log\localise.binlog
```

Comment:
```suggestion
    dotnet msbuild /p:Configuration=Release /t:_UpdateEnglishTranslations /p:RunTranslationApp=true $buildArgs /v:m /bl:..\artifacts\log\localise.binlog
```

Reply:
This wouldn't work as you expect it - `$buildArgs` don't get interpreted as input parameters:
![image](https://user-images.githubusercontent.com/4403806/211223642-9d65bd7c-b5fa-4012-b56f-7b391b904f45.png)
https://ci.appveyor.com/project/gitextensions/gitextensions/builds/45871046

So, the whole command needs to be encoded as a string and then executed.

Reply:
I specifically left this command as is - the translation verification step does not perform a build and it does not require commit hashes.